apps/emqx_gateway/src/emqx_gateway_api_authn.erl

```diff
@@ -90,63 +83,31 @@ authn(delete, #{bindings := #{name := Name0}}) ->
         {204}
     end).
 
-users(get, #{bindings := #{name := Name0}, query_string := Qs}) ->
-    with_authn(Name0, fun(
-        _GwName,
-        #{
-            id := AuthId,
-            chain_name := ChainName
-        }
-    ) ->
-        emqx_authn_api:list_users(ChainName, AuthId, parse_qstring(Qs))
-    end);
-users(post, #{
-    bindings := #{name := Name0},
-    body := Body
-}) ->
-    with_authn(Name0, fun(
-        _GwName,
-        #{
-            id := AuthId,
-            chain_name := ChainName
-        }
-    ) ->
-        emqx_authn_api:add_user(ChainName, AuthId, Body)
-    end).
+users(get, #{query_string := Qs} = Req) ->
+    #{gateway := #{id := AuthId, chain_name := ChainName}} = Req,
+    emqx_authn_api:list_users(ChainName, AuthId, parse_qstring(Qs));
+users(post, #{body := Body} = Req) ->
+    #{gateway := #{id := AuthId, chain_name := ChainName}} = Req,
+    emqx_authn_api:add_user(ChainName, AuthId, Body).
 
-users_insta(get, #{bindings := #{name := Name0, uid := UserId}}) ->
-    with_authn(Name0, fun(
-        _GwName,
-        #{
-            id := AuthId,
-            chain_name := ChainName
-        }
-    ) ->
-        emqx_authn_api:find_user(ChainName, AuthId, UserId)
-    end);
-users_insta(put, #{
-    bindings := #{name := Name0, uid := UserId},
-    body := Body
-}) ->
-    with_authn(Name0, fun(
-        _GwName,
-        #{
-            id := AuthId,
-            chain_name := ChainName
-        }
-    ) ->
-        emqx_authn_api:update_user(ChainName, AuthId, UserId, Body)
-    end);
-users_insta(delete, #{bindings := #{name := Name0, uid := UserId}}) ->
-    with_authn(Name0, fun(
-        _GwName,
-        #{
-            id := AuthId,
-            chain_name := ChainName
-        }
-    ) ->
-        emqx_authn_api:delete_user(ChainName, AuthId, UserId)
-    end).
+users_insta(get, #{bindings := #{uid := UserId}} = Req) ->
+    #{
+        gateway := #{id := AuthId, chain_name := ChainName},
+        resolved_ns := Namespace
```

Comment:
we don't support namespace for gateways for now. maybe hardcode it to global ns for now.